src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -593,6 +605,9 @@ void SpatialPooler::initialize(vector<UInt> inputDimensions,
     printParameters();
     std::cout << "CPP SP seed                 = " << seed << std::endl;
   }
+
+  //check for reasonable params
+  NTA_CHECK(checkUnstableParams_(*this)); //TODO the assert runs only at debug builds, make mandatory?
```

Comment:
TODO: make this mandatory even for Release builds?

Reply:
Currently the check crashes? on some instances, while already identifies some problematic instances in our unit-tests.

Reply:
..  crash fixed in #1391